### PR TITLE
update FullStory destination to v2.0

### PR DESCRIPTION
**Pull request by ScottLNorvell**
remove some debug stuff

additional cleanup

_A summary of your pull request, including the what change you're making and why._

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [ ] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

**Review by djwhatley-fs — COMMENTED** [review]
This looks pretty good to me, though I might not have the best context for it. Nothing jumps out, anyway.

**Comment by patrick-fs** [comment]
I think you'll also want to change this `presets` declaration to point to the `V2` actions: https://github.com/ScottLNorvell/action-destinations/blob/71aae5448ea67aa813a8a98f3a344a8aa2ec9fd8/packages/browser-destinations/destinations/fullstory/src/index.ts#L26

```
presets: [
    {
      name: 'Track Event', // becomes 'Track Event V2'
      subscribe: 'type = "track"',
      partnerAction: 'trackEvent', // becomes trackEventV2
      mapping: defaultValues(trackEvent.fields),
      type: 'automatic'
    },
    {
      name: 'Identify User', // becomes 'Identify User V2'
      subscribe: 'type = "identify"',
      partnerAction: 'identifyUser', // becomes identifyUserV2
      mapping: defaultValues(identifyUser.fields),
      type: 'automatic'
    }
  ],
  ```

**Comment by patrick-fs** [comment]
oh! and I think we'd want to add a preset for viewedPage:

```
...
    name: `Viewed Page`,
    subscribe: 'type = "page"',
    partnerAction: 'viewedPageV2',
...
```

File: packages/browser-destinations/destinations/fullstory/src/__tests__/fullstory.test.ts [hunk 1]
```diff
@@ -1,83 +1,41 @@
...
     enabled: true,
     subscribe: 'type = "track"',
-    mapping: {
-      name: {
-        '@path': '$.name'
-      },
-      properties: {
-        '@path': '$.properties'
-      }
-    }
```
Comment (ScottLNorvell) [inline packages/browser-destinations/destinations/fullstory/src/__tests__/fullstory.test.ts:18]: Here and on line 96 below, our default fields didn't match up with what we were testing previously. I wanted to confirm this was what we want as default values. We could always do something like what we do in viewedPage with the if statement here: https://github.com/ScottLNorvell/action-destinations/pull/1/files#diff-a44bbdd428f7e7319a3625d9db97d15cc9d01ff12373dcde8c6ca57faacf73c6R19-R23

**Review by ScottLNorvell — COMMENTED** [review]
Assigning to @patrick-fs for now. See the one question I have in the comments.

Also: Does it make sense to update the default server api integration to v2 or did we need to update the documentation first?

After feedback is received, I'll merge this pr into `main` of this fork and then submit a pull request from my fork. Does that seem like the right plan?

Thanks!

File: packages/browser-destinations/destinations/fullstory/src/identifyUserV2/index.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    }
+  },
+  perform: (FS, event) => {
+    let newTraits: Record<string, unknown> = {}
+
+    if (event.payload.traits) {
+      newTraits = Object.entries(event.payload.traits).reduce(
+        (acc, [key, value]) => ({
+          ...acc,
+          [camelCaseField(key)]: value
```
Comment (patrick-fs) [inline packages/browser-destinations/destinations/fullstory/src/identifyUserV2/index.ts]: I don't think we want to mess with field names for V2. WDYT?

**Comment by patrick-fs** [comment]
@ScottLNorvell I left a comment for you to review. Would you mind working with Nate on the rest of the PR/make him the assignee. He's happy to help and has worked on the Segment integration before.

File: packages/browser-destinations/destinations/fullstory/src/identifyUserV2/index.ts (hunk 2 above)
Comment (ScottLNorvell) [reply packages/browser-destinations/destinations/fullstory/src/identifyUserV2/index.ts]: Seems like something we could remove.

Comment (patrick-fs) [reply packages/browser-destinations/destinations/fullstory/src/identifyUserV2/index.ts]: Cool, please do!

File: packages/browser-destinations/destinations/fullstory/src/index.ts [hunk 3]
```diff
@@ -24,15 +27,22 @@ export const destination: BrowserDestinationDefinition<Settings, FS> = {
     {
       name: 'Track Event',
       subscribe: 'type = "track"',
-      partnerAction: 'trackEvent',
-      mapping: defaultValues(trackEvent.fields),
+      partnerAction: 'trackEventV2',
+      mapping: defaultValues(trackEventV2.fields),
```
Comment (ndonze-fs) [inline packages/browser-destinations/destinations/fullstory/src/index.ts:31]: Is the intent for v2 actions to immediately become the default for customers setting up new browser/device mode destinations? If so, this looks good. In contrast, when we rolled out cloud mode v2 actions we didn't immediately set those as the default. Instead, select customers were opted in as sort of a beta. /cc @patrick-fs

File: packages/browser-destinations/destinations/fullstory/src/identifyUserV2/index.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
+import type { BrowserActionDefinition } from '@segment/browser-destination-runtime/types'
+import type { Settings } from '../generated-types'
+import type { Payload } from './generated-types'
+import type { FS } from '../types'
+import { segmentEventSource } from '..'
+
+// Change from unknown to the partner SDK types
+const action: BrowserActionDefinition<Settings, FS, Payload> = {
+  title: 'Identify User',
```
Comment (ndonze-fs) [inline packages/browser-destinations/destinations/fullstory/src/identifyUserV2/index.ts]: In our cloud mode (server side) equivalents of these actions, we included the version number in the title e.g. "Identify User V2". Does it make sense to do the same here?

File: packages/browser-destinations/destinations/fullstory/src/viewedPageV2/index.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
+import type { BrowserActionDefinition } from '@segment/browser-destination-runtime/types'
+import type { Settings } from '../generated-types'
+import type { Payload } from './generated-types'
+import type { FS } from '../types'
+import { segmentEventSource } from '..'
+
+const action: BrowserActionDefinition<Settings, FS, Payload> = {
+  title: 'Viewed Page',
```
Comment (ndonze-fs) [inline packages/browser-destinations/destinations/fullstory/src/viewedPageV2/index.ts]: Similar question about including a version in the title e.g. "Viewed Page V2"

File: packages/browser-destinations/destinations/fullstory/src/trackEventV2/index.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
+import type { BrowserActionDefinition } from '@segment/browser-destination-runtime/types'
+import type { Settings } from '../generated-types'
+import type { Payload } from './generated-types'
+import type { FS } from '../types'
+import { segmentEventSource } from '..'
+
+const action: BrowserActionDefinition<Settings, FS, Payload> = {
+  title: 'Track Event',
```
Comment (ndonze-fs) [inline packages/browser-destinations/destinations/fullstory/src/trackEventV2/index.ts]: Similar question about including a version in the title e.g. "Track Event V2"

File: packages/browser-destinations/destinations/fullstory/src/__tests__/fullstory.test.ts [hunk 7]
```diff
@@ -236,3 +194,93 @@ describe('#identify', () => {
     )
   })
 })
+
+describe('#page', () => {
+  it('sends page events to fullstory on "page" (category edition)', async () => {
+    const [, , viewed] = await fullstory({
+      orgId: 'thefullstory.com',
```
Comment (ndonze-fs) [inline packages/browser-destinations/destinations/fullstory/src/__tests__/fullstory.test.ts]: Nit (no strong feelings about this): Wonder if it's worth using a fake org id given this code ends up in Segment's publicly visible OSS repo.

File: packages/browser-destinations/destinations/fullstory/src/__tests__/fullstory.test.ts [hunk 8]
```diff
@@ -236,3 +194,93 @@ describe('#identify', () => {
...
+      'page',
+      {
+        pageName: 'Walrus Page',
+        banana: '📞'
+      },
+      'segment-browser-actions'
+    )
+  })
+
+  it('sends page events to fullstory on "page" (no pageNaem edition)', async () => {
```
Comment (ndonze-fs) [inline packages/browser-destinations/destinations/fullstory/src/__tests__/fullstory.test.ts]:
```suggestion
  it('sends page events to fullstory on "page" (no pageName edition)', async () => {
```

**Review by ndonze-fs — APPROVED** [review]
Left a couple questions and small nits. Otherwise LGTM.

File: packages/browser-destinations/destinations/fullstory/src/index.ts (hunk 3 above)
Comment (patrick-fs) [reply packages/browser-destinations/destinations/fullstory/src/index.ts:31]: Yes, I think we should automatically default new customer onto V2.

File: packages/browser-destinations/destinations/fullstory/src/__tests__/fullstory.test.ts (hunk 7 above)
Comment (patrick-fs) [reply packages/browser-destinations/destinations/fullstory/src/__tests__/fullstory.test.ts]: +1

**Comment by patrick-fs** [comment]
@ndonze-fs would it be a good idea to include changes to the cloud mode defaults in the PR? It reduces the number of PRs we have to take through Segment review, and it's planned work anyway.

**Comment by ndonze-fs** [comment]
> @ndonze-fs would it be a good idea to include changes to the cloud mode defaults in the PR? It reduces the number of PRs we have to take through Segment review, and it's planned work anyway.

SGTM. We could do that as a separate internal PR in this same repo and have one upstream PR for Segment to review.

File: packages/browser-destinations/destinations/fullstory/src/identifyUserV2/index.ts (hunk 4 above)
Comment (ScottLNorvell) [reply packages/browser-destinations/destinations/fullstory/src/identifyUserV2/index.ts]: Totally makes sense. This is probably the display value to the user.  themoreyouknow

**Comment by ScottLNorvell** [comment]
> SGTM. We could do that as a separate internal PR in this same repo and have one upstream PR for Segment to review.

blob-sgtm

I'm making those few changes. Then I'll make a separate PR against my main for the cloud mode defaults.

File: packages/browser-destinations/destinations/fullstory/src/__tests__/fullstory.test.ts (hunk 8 above)
Comment (ScottLNorvell) [reply packages/browser-destinations/destinations/fullstory/src/__tests__/fullstory.test.ts]: #speling

**Pull request merged**